src/components/app/app.scss

```diff
@@ -3,23 +3,45 @@
 //--- Reusable site-wide styles --//
 
 .pipeline-icon {
+  transition: fill 300ms ease, stroke 300ms ease;
+
+  .kui-theme--light & {
+    fill: rgba($color-dark, 0.7);
+  }
+
+  .kui-theme--dark & {
+    fill: rgba($color-light, 0.7);
+  }
+
+  .kui-theme--light .pipeline-icon-toolbar__button:hover & {
+    fill: $color-dark;
+  }
+
+  .kui-theme--dark .pipeline-icon-toolbar__button:hover & {
+    fill: $color-light;
+  }
+}
+
+.pipeline-icon--stroke .pipeline-icon {
```

Comment:
Yeah I don't think I named this well originally as part of the minimap PR, so I've refactored this as suggested, thanks.

Reply:
I see that it's now 
```css
.pipeline-icon.pipeline-icon--stroke {
```
Could this not just be a single class?
```css
.pipeline-icon--stroke {
```
Why does it need the additional class? As mentioned previously, the whole point of using BEM is to avoid needing to compound the classes in order to avoid the increased specificity. It's not clear from reading it that the additional specificity is necessary.

Reply:
This was needed as the default selector had higher specificity due to properties being set through theme classes. I've instead duplicated the `fill` property here to both theme classes instead so this can work as suggested.